### PR TITLE
feat(query): Allow custom AsyncHttpClientConfig in RemoteExec

### PR DESCRIPTION
**Pull request by tjackpaul**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

### CODE REVIEW

File: coordinator/src/test/scala/filodb.coordinator/queryplanner/HighAvailabilityPlannerSpec.scala [hunk 1]
```diff
@@ -31,8 +31,8 @@ class HighAvailabilityPlannerSpec extends AnyFunSpec with Matchers {
   private val dsRef = dataset.ref
   private val schemas = Schemas(dataset.schema)
 
-  private val routingConfigString = "routing {\n  remote {\n   " +
-    " http {\n      endpoint = localhost\n timeout = 10000\n    }\n  }\n}"
+  private val routingConfigString = "routing {\n  remote {\n    http {\n      endpoint = localhost\n      timeout = 10000\n" +
+    "      client.sttp-backend.factory = filodb.query.exec.AsyncHttpClientBackendFactory\n    }\n  }\n}"
```
Comment (vishramachandran) [inline coordinator/src/test/scala/filodb.coordinator/queryplanner/HighAvailabilityPlannerSpec.scala:35]: Use `"""` format for multi line strings

File: query/src/main/scala/filodb/query/exec/PromQlRemoteExec.scala [hunk 2]
```diff
@@ -114,7 +44,7 @@ case class PromQlRemoteExec(queryEndpoint: String,
 
   override def sendHttpRequest(execPlan2Span: Span, httpTimeoutMs: Long)
                               (implicit sched: Scheduler): Future[QueryResponse] = {
-    PromRemoteExec.httpGet(queryEndpoint, requestTimeoutMs, queryContext.submitTime, getUrlParams())
+    RemoteHttpClient(sttpBackend).httpGet(queryEndpoint, requestTimeoutMs, queryContext.submitTime, getUrlParams())
```
Comment (vishramachandran) [inline query/src/main/scala/filodb/query/exec/PromQlRemoteExec.scala:47]: Avoid creation of RemoteHttpClient multiple times

**Review by vishramachandran — APPROVED** [review]
Approved with suggestions

File: coordinator/src/test/scala/filodb.coordinator/queryplanner/HighAvailabilityPlannerSpec.scala (hunk 1 above)
Comment (tjackpaul) [reply coordinator/src/test/scala/filodb.coordinator/queryplanner/HighAvailabilityPlannerSpec.scala]: This change is no more required. Reverted

File: query/src/main/scala/filodb/query/exec/RemoteExec.scala [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+/**
+ * A zero-arg constructor class that knows how to create an HttpClient for Prom Remote Queries.
+ */
+trait RemoteExecSttpBackendFactory {
+
+  /**
+   * Returns an SttpBackend that can be used to create sttp for remote Http requests.
+   * @param config the configuration for the http client
+   */
+  def create(config: Config): SttpBackend[Future, Nothing]
```
[the remark this answers is not in the document]
Comment (tjackpaul) [reply query/src/main/scala/filodb/query/exec/RemoteExec.scala:98]: Yes, will go with this existing Sttp client for now. We will abstract this out later if required.

**Review by TanviBhavsar — APPROVED** [review]

**Pull request merged**